src/types/route.ts

```diff
@@ -1,17 +1,19 @@
 import { ExtractParamTypes, MergeParams, Param } from '@/types/params'
-import { RouteProps } from '@/types/routeProps'
+import { RouteMeta, RouteProps } from '@/types/routeProps'
 import { Path, Query, ToPath, ToQuery } from '@/utilities'
 
 export type Routes = Readonly<Route[]>
 
+type RoutePropsWithMeta = RouteProps & { meta: RouteMeta }
```

Comment:
Why isn't meta just part of RouteProps?

Reply:
it's optional in `RouteProps` and should be

Reply:
Oh right yep that makes sense. Yay another route type 🤪😂

Reply:
lol this one is not exported so it doesn't count

Reply:
Haha I can get behind that philosophy